internal/orchestrator/updates.go

```diff
@@ -35,6 +35,9 @@ var (
 	errCompleteEvent   = errors.New("unable to complete event")
 	failedByReconciler = []byte(`{ "msg": "controller failed to process this condition in time" }`)
 	reconcilerCadence  = 1 * time.Minute
+	// Ideally set this to the JS MaxAge value.
+	// The Discard Policy should be set to Old.
+	msgMaxAgeThreshold = 24 * time.Hour
```

Comment:
We can read this from `nats.StreamConfig` via `nats.GetStreamInfo(...).Config.MaxAge`.

In re: discard policy, I don't know that it makes that much difference, but I'm open to the idea.

Reply:
Yeah, this should ideally be from the stream configuration we set, as of now `MaxAge` is unlimited.
In a followup I'll look into picking it up from the configuration instead.

The Discard Policy is currently set to `New`, with `MaxAge` defined it would throw errors when accepting new messages which would not be something we like I guess - we can discuss this offline.

https://docs.nats.io/nats-concepts/jetstream/streams#discardpolicy